console/src/schematic/Schematic.tsx

```diff
@@ -233,25 +254,18 @@ export const Loaded: Layout.Renderer = ({ layoutKey, visible }) => {
           ),
           0,
         );
-        dispatch(
+        undoableDispatch(
           addElement({
             key: layoutKey,
             elKey: id.id(),
-            node: {
-              position: pos,
-              zIndex: spec.zIndex,
-            },
-            props: {
-              key,
-              ...spec.defaultProps(theme),
-              ...(data ?? {}),
-            },
+            node: { position: pos, zIndex: spec.zIndex },
+            props: { key, ...spec.defaultProps(theme), ...(data ?? {}) },
           }),
         );
       });
       return valid;
     },
-    [schematic.viewport, theme],
+    [theme, undoableDispatch, layoutKey, ref.current, viewportRef.current],
```

Comment:
`ref.current` and `viewportRef.current` should not be in the dependencies array.